src/components/img/QImg.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <div class="q-img">
+    <!-- trigger load on top of image, not bottom -->
+    <!-- TODO: check horizontal scroll -->
+    <span v-scroll-fire="() =>  { if (state === 0) state += 1 }"/>
+    <!-- this trnasition mostly optional, removes browser flicker -->
+    <transition name="preview">
+      <img
+        v-if="placeholder && state < 2"
+        :src="placeholder"
+        class="preview"
+        :alt="$attrs.alt"
+      >
+    </transition>
+    <transition name="reveal" @after-enter="state += 1">
+      <img
+        v-if="state > 0"
+        v-show="state > 1"
+        @load="state += 1"
```

Comment:
Is this supposed to be capturing the native onload event?

Reply:
Yes. Reference [here](http://jsfiddle.net/nobu222/xr0g3Lco/)

Reply:
Neat, I don't think I would have thought to have done that. If this event handlers were changed to:
`@after-enter="state = 3"`
`@load="state = 2"`
That'd be accurate, right?  And more explicit?

Reply:
Done